pkg/virt-controller/watch/vmi.go

```diff
@@ -307,24 +307,49 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 		}
 	}
 
+	hasWffcDataVolume := false
+	// there is no reason to check for waitForFirstConsumer is there are failed DV's
+	if !hasFailedDataVolume {
+		for _, dataVolume := range dataVolumes {
+			if dataVolume.Status.Phase == cdiv1.WaitForFirstConsumer {
+				hasWffcDataVolume = true
+			}
+		}
+	}
+
 	conditionManager := controller.NewVirtualMachineInstanceConditionManager()
 	vmiCopy := vmi.DeepCopy()
-	podExists := podExists(pod)
+	vmiPodExists := podExists(pod) && !isTempPod(pod)
+	tempPodExists := podExists(pod) && isTempPod(pod)
 
 	vmiCopy, err := c.setActivePods(vmiCopy)
 	if err != nil {
 		return fmt.Errorf("Error detecting vmi pods: %v", err)
 	}
 
 	switch {
-
 	case vmi.IsUnprocessed():
-		if podExists {
+		if vmiPodExists {
 			vmiCopy.Status.Phase = virtv1.Scheduling
 		} else if vmi.DeletionTimestamp != nil {
 			vmiCopy.Status.Phase = virtv1.Failed
 		} else if hasFailedDataVolume {
 			vmiCopy.Status.Phase = virtv1.Failed
+		} else if hasWffcDataVolume && !tempPodExists {
+			condition := virtv1.VirtualMachineInstanceCondition{
+				Type:   virtv1.VirtualMachineInstanceProvisioning,
+				Status: k8sv1.ConditionTrue,
+			}
+			cm := controller.NewVirtualMachineInstanceConditionManager()
+			if cm.HasCondition(vmiCopy, condition.Type) {
+				cm.RemoveCondition(vmiCopy, condition.Type)
+			}
+			vmiCopy.Status.Conditions = append(vmiCopy.Status.Conditions, condition)
+		} else if !hasWffcDataVolume && tempPodExists {
+			cm := controller.NewVirtualMachineInstanceConditionManager()
+			if cm.HasCondition(vmiCopy, virtv1.VirtualMachineInstanceProvisioning) {
+				cm.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceProvisioning)
+			}
```

Comment:
These if statements short circuit the final `else` below, which both sets the vmi phase to `pending` and reports any potential sync errors as a condition.

Regardless of the wffc logic, we still need to make sure sync errors are visible as a condition.